libs/island-ui/core/src/lib/ActionCard/ActionCard.tsx

```diff
@@ -129,27 +130,53 @@ export const ActionCard: React.FC<ActionCardProps> = ({
       ? 'red100'
       : 'blue100'
 
-  const renderAvatar = () => {
-    if (!avatar) {
+  const renderImage = () => {
+    if (!image) {
       return null
     }
 
-    return heading ? (
-      <Box
-        display="flex"
-        justifyContent="center"
-        alignItems="center"
-        flexShrink={0}
-        marginRight={[2, 3]}
-        borderRadius="circle"
-        background="blue100"
-        className={styles.avatar}
-      >
-        <Text variant="h3" as="p" color="blue400">
-          {getTitleAbbreviation(heading)}
-        </Text>
-      </Box>
-    ) : null
+    if (image.type === 'image') {
+      if (!image.url || image.url.length === 0) return null
+      return (
+        <Box
+          display="flex"
+          justifyContent="center"
+          alignItems="center"
+          flexShrink={0}
+          marginRight={[2, 3]}
+          borderRadius="circle"
+        >
+          <img className={styles.avatar} src={image.url} alt="action-card" />
+        </Box>
+      )
+    } else if (image.type === 'avatar') {
+      return heading ? (
+        <Box
+          display="flex"
+          justifyContent="center"
+          alignItems="center"
+          flexShrink={0}
+          marginRight={[2, 3]}
+          borderRadius="circle"
+          background="blue100"
+          className={styles.avatar}
+        >
+          <Text variant="h3" as="p" color="blue400">
+            {getTitleAbbreviation(heading)}
+          </Text>
+        </Box>
+      ) : null
+    } else if (image.type === 'logo') {
+      if (!image.url || image.url.length === 0) return null
+      return (
+        <Box
+          padding={2}
+          marginRight={2}
+          className={styles.logo}
+          style={{ backgroundImage: `url(${image.url})` }}
+        ></Box>
+      )
+    }
```

Comment:
We could rearrange conditions to minimize nested if statements and make it a little more readable 🙂
e.g.
```suggestion
  if (image.type === 'avatar' && heading) {
    return (
      <Box
        display="flex"
        justifyContent="center"
        alignItems="center"
        flexShrink={0}
        marginRight={[2, 3]}
        borderRadius="circle"
        background="blue100"
        className={styles.avatar}
      >
        <Text variant="h3" as="p" color="blue400">
          {getTitleAbbreviation(heading)}
        </Text>
      </Box>
    )
  }
  if (!image.url || image.url.length === 0) return null
  if (image.type === 'image') {
    return (
      <Box
        display="flex"
        justifyContent="center"
        alignItems="center"
        flexShrink={0}
        marginRight={[2, 3]}
        borderRadius="circle"
      >
        <img className={styles.avatar} src={image.url} alt="action-card" />
      </Box>
    )
  }
  if (image.type === 'logo') {
    return (
      <Box
        padding={2}
        marginRight={2}
        className={styles.logo}
        style={{ backgroundImage: `url(${image.url})` }}
      ></Box>
    )
  }
  return null
```

Reply:
Nice thanks 💯